### DISCUSSION TITLE
How to determine if a PR can be merged?

### DISCUSSION THREAD

**Discussion by xtough** [community/community#24504] [Apps, API and Webhooks]
<p>Hi folks,<br>
I am looking for a programmatic possibility to determine via v3 API  if a PR can be merged, i.e., actually the boolean status that makes the “merge” button green or red.</p>
<p>For instance, <a href="https://github.community/t/pullrequest-mergeable-state-possible-values/13926" class="inline-onebox">PullRequest mergeable_state possible values</a> goes in a similar direction. “mergeable” only let’s me know if the PR has no potential merge conflicts. But all the other criteria like reviews, statuses, etc. would need to be queried individually.</p>
<p>In v4 MergeStateStatus seems to be what I am looking for. But can I also achieve the same by v3 means?</p>
<p>Thanks,<br>
Christoph</p>

**Comment by francisfuzz** [comment, marked as answer]
<p>:wave: hello there and welcome to the GitHub Support Community, <a class="mention" href="/xtough">@xtough</a> !</p>
<aside class="quote no-group" data-username="xtough" data-post="1" data-topic="122190">
<div class="title">
<div class="quote-controls"></div>
 xtough:</div>
<blockquote>
<p>I am looking for a programmatic possibility to determine via v3 API if a PR can be merged, i.e., actually the boolean status that makes the “merge” button green or red.</p>
</blockquote>
</aside>
<aside class="quote no-group" data-username="xtough" data-post="1" data-topic="122190">
<div class="title">
<div class="quote-controls"></div>
 xtough:</div>
<blockquote>
<p>In v4 MergeStateStatus seems to be what I am looking for. But can I also achieve the same by v3 means?</p>
</blockquote>
</aside>
<p>I recently <a href="https://github.community/t/how-can-i-check-the-state-of-the-require-branches-to-be-up-to-date-before-merging-check-for-a-pull-request/122782/2">answered a similar question here</a>. I’m wondering if that helps with what you’re looking for? :thought_balloon:</p>

**Reply by Maxim-Mazurok** [reply]
The link is broken ;(

**Comment by xtough** [comment]
<p>Thanks francisfuzz. Yes, this sort of helps. mergeable_state==“BLOCKED” is indeed what I was looking for. I was just confused because it is not part of the documented API. I guess the chances are small that the possible values will change incompatibly, no that the v4 API is more or less cementing the same.</p>

**Comment by Maxim-Mazurok** [comment]
This seems relevant: [docs.github.com/en/rest/guides/using-the-rest-api-to-interact-with-your-git-database?apiVersion=2022-11-28#checking-mergeability-of-pull-requests](https://docs.github.com/en/rest/guides/using-the-rest-api-to-interact-with-your-git-database?apiVersion=2022-11-28#checking-mergeability-of-pull-requests)

> Warning! Please do not depend on using Git directly or [GET /repos/{owner}/{repo}/git/refs/{ref}](https://docs.github.com/en/rest/git#get-a-reference) for updates to merge Git refs, because this content becomes outdated without warning.
> 
> A consuming API needs to explicitly request a pull request to create a test merge commit. A test merge commit is created when you view the pull request in the UI and the "Merge" button is displayed, or when you [get](https://docs.github.com/en/rest/pulls#get-a-pull-request), [create](https://docs.github.com/en/rest/pulls#create-a-pull-request), or [edit](https://docs.github.com/en/rest/pulls#update-a-pull-request) a pull request using the REST API. Without this request, the merge Git refs will fall out of date until the next time someone views the pull request.
> 
> If you are currently using polling methods that produce outdated merge Git refs, then GitHub recommends using the following steps to get the latest changes from the default branch:
> 
> Receive the pull request webhook.
> Call [GET /repos/{owner}/{repo}/pulls/{pull_number}](https://docs.github.com/en/rest/pulls#get-a-pull-request) to start a background job for creating the merge commit candidate.
> Poll your repository using [GET /repos/{owner}/{repo}/pulls/{pull_number}](https://docs.github.com/en/rest/pulls#get-a-pull-request) to see if the mergeable attribute is true or false. You can use Git directly or [GET /repos/{owner}/{repo}/git/refs/{ref}](https://docs.github.com/en/rest/git#get-a-reference) for updates to merge Git refs only after performing the previous steps.
